Mathlib/Data/Matrix/Rank.lean

```diff
@@ -224,8 +224,9 @@ theorem ker_mulVecLin_conjTranspose_mul_self (A : Matrix m n R) :
     replace h := congr_arg (dotProduct (star x)) h
     haveI : NoZeroDivisors R := inferInstance
     rwa [dotProduct_mulVec, dotProduct_zero, vecMul_conjTranspose, star_star,
```

Comment:
I think most of this proof can be replaced with `conj_transpose_mul_self_mul_eq_zero_iff`?

Reply:
(or perhaps the mulVec version that I suggested you add)

Reply:
I am not following here. This does what `conj_transpose_mul_self_mul_eq_zero_iff` but for vectors instead of full matrices. Which are you trying to replace?

Reply:
Sorry for the confusion, what I'm proposing is:
* Address https://github.com/leanprover-community/mathlib4/pull/6587#discussion_r1295590590, either by extracting the proof here or re-proving from scratch.
* Golf the proof here to use the new lemma.